rotkehlchen/chain/ethereum/adex/adex.py

```diff
@@ -368,22 +364,34 @@ def _deserialize_bond(
     ) -> Bond:
         """Deserialize a bond event.
 
-        It may raise KeyError.
+        May raise DeserializationError.
         """
-        adex_event = self._deserialize_adex_event(
-            raw_event=raw_event,
-            identity_address_map=identity_address_map,
-        )
-        amount_int = int(raw_event['amount'])
-        amount = FVal(raw_event['amount']) / ADX_AMOUNT_MANTISSA
-        pool_id = raw_event['poolId']
-        nonce = int(raw_event['nonce'])
-        bond_id = self._get_bond_id(
-            identity_address=adex_event.identity_address,
-            amount=amount_int,
-            pool_id=pool_id,
-            nonce=nonce,
-        )
+        try:
+            adex_event = self._deserialize_adex_event(
+                raw_event=raw_event,
+                identity_address_map=identity_address_map,
+            )
+            amount_int = int(raw_event['amount'])
+            amount = FVal(raw_event['amount']) / ADX_AMOUNT_MANTISSA
+            pool_id = raw_event['poolId']
+            nonce = int(raw_event['nonce'])
+            bond_id = self._get_bond_id(
+                identity_address=adex_event.identity_address,
+                amount=amount_int,
+                pool_id=pool_id,
+                nonce=nonce,
+            )
+            slashed_at = deserialize_timestamp(raw_event['slashedAtStart'])
+        except (DeserializationError, KeyError, ValueError) as e:
```

Comment:
Deserializing the events I call several times `deserialize_timestamp`. I've included it here as I can't see a better way to simplify the messages the user will see (1 per failing event at max and specifying the event "type") and to avoid duplicated blocks of code in the error handling (e.g. capture the DeserializationError in particular and the user-understandable reason).

Reply:
are you referring to catching and re-raising DeserializationEror?

Reply:
yes, I meant catching and re-raising it with a standardised message